app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt

```diff
@@ -0,0 +1,262 @@
+  // Reference to the 'Users' collection in Firestore
+  private lateinit var usersCollection: CollectionReference
+  // Reference to the 'Trips' collection in Firestore
+  private lateinit var tripsCollection: CollectionReference
+
+  /**
+   * (Currently used only for unit tests)
+   *
+   * Initializes Firestore with a specified FirebaseApp. Sets up 'Users' and 'Trips' collection
+   * references. Use when working with a specific FirebaseApp instance.
+   *
+   * @param app The FirebaseApp instance for Firestore initialization.
+   */
+  fun initFirestore(app: FirebaseApp) {
+    firestore = FirebaseFirestore.getInstance(app)
+    usersCollection = firestore.collection("Users")
+    tripsCollection = firestore.collection("Trips")
+  }
+
+  /**
+   * Initializes Firestore with the default FirebaseApp. Sets up 'Users' and 'Trips' collection
+   * references. Use for applications with a single Firebase project.
+   */
+  fun initFirestore() {
+    firestore = FirebaseFirestore.getInstance()
+    usersCollection = firestore.collection("Users")
+    tripsCollection = firestore.collection("Trips")
+  }
+
+  /**
+   * Asynchronously retrieves a trip by its ID from Firestore and converts it to the data model.
+   *
+   * @param tripId The unique identifier of the trip to retrieve.
+   * @return The Trip object if found, null otherwise.
+   */
+  suspend fun getTrip(tripId: String): Trip? =
+      withContext(Dispatchers.IO) {
+        try {
+          val documentSnapshot = tripsCollection.document(tripId).get().await()
+          val firestoreTrip =
+              documentSnapshot.toObject<
+                  FirestoreTrip>() // Converts Firestore document to FirestoreTrip DTO
+          firestoreTrip?.toTrip() // Converts FirestoreTrip DTO to Trip data model
+        } catch (e: Exception) {
+          null // error
+        }
+      }
+
+  /**
+   * Fetches multiple trips by their IDs, leveraging the getTrip function for each ID.
+   *
+   * @param tripIds List of trip IDs to fetch.
+   * @return List of Trip objects.
+   */
+  suspend fun getAllTrips(tripIds: List<String>): List<Trip> =
+      withContext(Dispatchers.IO) {
+        tripIds.mapNotNull { tripId ->
+          try {
+            getTrip(tripId) // Utilizes the getTrip method to fetch each trip individually
+          } catch (e: Exception) {
+            null // error
+          }
+        }
+      }
+
+  /**
+   * Adds a new trip to Firestore, converting the Trip data model to FirestoreTrip DTO for storage.
+   *
+   * @param trip The Trip object to add.
+   * @return Boolean indicating success or failure of the operation.
+   */
+  suspend fun addTrip(trip: Trip): Boolean =
+      withContext(Dispatchers.IO) {
+        try {
+          // Generate a unique ID for the trip
+          var uniqueID = UUID.randomUUID().toString()
+
+          var existingDocument = tripsCollection.document(uniqueID).get().await()
+          while (existingDocument
```

Comment:
This is maybe a bit overkill since UUID guarantess uniqueness of IDS. The probability to get a colision probably shouldn't being considered according to this project